staging/src/kubevirt.io/api/snapshot/v1alpha1/types.go

```diff
@@ -91,6 +91,7 @@ const (
 	Succeeded  VirtualMachineSnapshotPhase = "Succeeded"
 	Failed     VirtualMachineSnapshotPhase = "Failed"
 	Deleting   VirtualMachineSnapshotPhase = "Deleting"
+	Errored    VirtualMachineSnapshotPhase = "ErrorState"
```

Comment:
Unfortunately, this is problematic in terms of backward compatibility with older clients.

We have a combination of two points that makes it problematic:
- The use of a `phase` is discouraged [1] due to its state machine interpretation. Older clients are not aware of new added states, therefore they may handle new states as an exception path. While it may be ok in some cases, the general default is to avoid it and use conditions instead.
- When discussing the use of API enums [2], similar reasoning may apply even for a different usage than the one mentioned before.  If the value is expected to be interpreted as informative with no ties to a specific state machine, then it is safe for use.

[1] https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md?plain=1#L495
[2] https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#new-enum-value-in-existing-field

I would recommend to leave this information as part of the condition and instruct clients to query for the condition in order to check if there is an error state.
Avoid changing the existing phases and possibly mark it for deprecation (but not removal) to avoid clients for depending on it.

Reply:
@EdDev ok after some thought I agree in this case that the added value is no great since we have the conditions and the 'Error' in the status to reflect that. I added another commit to remove, please take a look

Reply:
Thank you.

It will be nice if the last commit will be squashed in this first commit, to avoid adding and removing (it is a bit confusing).